drivers: clock_control: Added PLL fractional mode for STM32U5

jandriea opened the pull request:
Based on RM0456, each PLL in the STM32U5 has the capability to work either in integer or fractional mode. In this update, the fractional mode can be enabled by setting the `fracn` value in the device tree. This update also includes a test case that generates a 160 MHz system clock using a 16777216 Hz HSE clock.

gautierg-st reviewed (changes requested):
Please split in several commits, at least the test part.
Otherwise looks good to me.

jandriea committed:
drivers: clock_control: Add PLL fracn for STM32U5

Based on RM0456, each PLL in the STM32U5 has the
capability to work either in integer or fractional mode.
In this update, the fractional mode can be enabled
by setting the fracn value in the device tree.

Signed-off-by: Jatty Andriean <jandriea@outlook.com>

erwango commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core/boards/pll_hse_fracn_160.overlay:
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
```
Please use your own copyright.

jandriea replied:
Thanks for the reminder, I will change the copyright later.

erwango commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core/boards/pll_hse_fracn_160.overlay:
```diff
@@ -0,0 +1,40 @@
+
+&clk_hse {
+	status = "okay";
+	clock-frequency = <16777216>;
```
IIUC, PLL frac doesn't have impact on PLL source but PLL output.

jandriea replied:
correct, the PLL frac only has an impact on the PLL output. In this test case, I want to generate a 160 Mhz PLL output using this 16777216 Hz HSE value which requires the PLL to be in the fractional mode.

erwango commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core/boards/pll_hse_fracn_160.overlay:
```diff
@@ -0,0 +1,40 @@
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(160)>;
```
This is the freq I'd expect to be modified

jandriea replied:
I am not quite sure about this. From my understanding, the RCC will use PLL as the clock source with a 160 MHz value. The 160 MHz value is the PLL output that was generated using the 16777216 Hz HSE. 

Please let me know if I am missing something.

erwango replied:
Ok, I was the one confused. Fine for me, thanks

erwango commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core/testcase.yaml:
```diff
@@ -24,3 +24,7 @@ tests:
     extra_args: DTC_OVERLAY_FILE="boards/clear_clocks.overlay;boards/pll_hse_160.overlay"
     # Build only as HSE not implemened on available boards
     build_only: true
+  drivers.stm32_clock_configuration.u5.sysclksrc_pll_hse_fracn_160:
```
What about using HSI as PLL input which would allow a "runable" test ?

jandriea replied:
I can add a test case using HSI with a fraction value of 0. Just to make sure that it still generates the correct PLL output. I will think about any other cases later. But for now, I can only suggest using HSI with a fraction value of 0. Perhaps, you have some test cases in mind?

erwango replied:
Let's go with 0

erwango reviewed (changes requested):
Otherwise LGTM

jandriea committed:
tests: drivers: clock_control: Add PLL fracn test

Added a test case that generates a 160 MHz system clock
using a 16777216 Hz HSE clock and also using a 16 MHz HSI

Signed-off-by: Jatty Andriean <jandriea@outlook.com>
